File: LogUtil.js
class LogUtil {

    static #now() {
        const d = new Date();
        const y = d.getFullYear();
        const M = String(d.getMonth() + 1).padStart(2, '0');
        const D = String(d.getDate()).padStart(2, '0');
        const H = String(d.getHours()).padStart(2, '0');
        const m = String(d.getMinutes()).padStart(2, '0');
        const s = String(d.getSeconds()).padStart(2, '0');
        return `${y}-${M}-${D}-${H}-${m}-${s}`;
    }

    static #format(msg) {
        if (typeof msg === 'object' && msg !== null) {
            return '\n' + JSON.stringify(msg, null, 2);
        }
        return msg;
    }

    static info(msg) {
        console.log(`[${this.#now()}] [INFO] ${this.#format(msg)}`);
    }

    static warn(msg) {
        console.log(`[${this.#now()}] [WARN] ${this.#format(msg)}`);
    }

    static error(msg, err) {
        let errStack = err?.stack ? '\n' + err.stack : '';
        console.error(`[${this.#now()}] [ERROR] ${this.#format(msg)}${errStack}`);
    }

    static debug(msg) {
        console.log(`[${this.#now()}] [DEBUG] ${this.#format(msg)}`);
    }
}

module.exports = LogUtil;

File: LogUtil.py
import json
import traceback
from datetime import datetime

class LogUtil:

    @staticmethod
    def _now():
        return datetime.now().strftime("%Y-%m-%d-%H-%M-%S")

    @staticmethod
    def _format(msg):
        if isinstance(msg, (dict, list)):
            return "\n" + json.dumps(msg, ensure_ascii=False, indent=2)
        return str(msg)

    @staticmethod
    def info(msg):
        print(f"[{LogUtil._now()}] [INFO] {LogUtil._format(msg)}")

    @staticmethod
    def warn(msg):
        print(f"[{LogUtil._now()}] [WARN] {LogUtil._format(msg)}")

    @staticmethod
    def error(msg, e=None):
        err_str = ""
        if e:
            err_str = "\n" + traceback.format_exc()
        print(f"[{LogUtil._now()}] [ERROR] {LogUtil._format(msg)}{err_str}")

    @staticmethod
    def debug(msg):
        print(f"[{LogUtil._now()}] [DEBUG] {LogUtil._format(msg)}")
